### ISSUE TITLE
Meetup September 5th

### ISSUE THREAD

**Issue by badboy** [Rustaceans/rust-cologne#6]
With #5 and the August meetup finished, let's get on to September:
- [x] Send "Save the Date" info
- [x] Final invitation (deadline Aug 29)
  - [x] meetup
  - [x] website
  - [x] twitter
  - [x] community calendar
  - [x] reddit
  - [x] rust-lang forum
  - [x] rustplatz
- [x] Agenda planing

**Comment by killercup** [comment]
Talk suggestions (don't announce just yet, can be recycled in September):
- @badboy about compiling to asm.js (test-talk for RustBeltRust IIRC)
- me about diesel
- that fancy new future-rs and tokio
- me about building that builder pattern macro of doom 😉

Jan-Erik Rediger notifications@github.com schrieb am Mi. 3. Aug. 2016 um
23:26:

**Comment by colin-kiegel** [comment]
I'll be in Utrech on September 7th. Sometime ago we decided, that it doesn't need to always be Wednesdays. => So, by any chance, would you all be ok with Monday Sept 5th instead?
- @Florob: Is the C4 conference room available on that date?

I really liked the evening yesterday - it was all very spontaneous and still had a "red line". I think we could try this again, i.e. pick a general topic for the evening and improvise more or less + optional talk. The general topic could be something broad as "storing / retrieving data with rust" with a focus on diesel (or alternative broad topic which would also fit the diesel-talk: "the emerging rust web stack").

Speaking of topic - I find all topic suggestions of @killercup super interesting. If @badboy would like to have a test talk about compiling to asm.js, I think we can prioritize that one a bit. My next pick would be diesel or some more macros right now.

BTW: Yesterday, someone suggested, we could try something like [open space](https://de.wikipedia.org/wiki/Open_Space) - however wikipedia suggests that it is suitable for 50+ participants. His feedback was, that he enjoyed the evening and in particular the individual discussions. But from a beginners perspective it was a bit difficult to follow the more advanced discussions in the group. In his opinion the open space concept could help to pair groups with more similar discussion interests.

EDIT: I just realized I unintentionally managed to mention all of you in one comment. ^^

**Comment by colin-kiegel** [comment]
I suggest to link this disccusion in our save-the-date invitation - how do you feel about this?

**Comment by killercup** [comment]
>  So, by any chance, would you all be ok with Monday Sept 5th instead?

Works for me. The week after that is ViewSource and RustFest, so I'll be in Berlin.

> I suggest to link this disccusion in our save-the-date invitation - how do you feel about this?

Sounds good. We should send this out as soon as we settled on a date.

> EDIT: I just realized I unintentionally managed to mention all of you in one comment. ^^

Nice :) You can also do "@​​Rustaceans/organizers-cologne" (← no link thanks to a zero-width space!)

**Comment by Florob** [comment]
The 5th should be fine, i'll try to get an official okay next Tuesday.

**Comment by badboy** [comment]
5th works for me as well. I'd appreciate if I can test out some of the things I plan for Rust Belt Rust (then I have the pressure to finish something by then)

**Comment by colin-kiegel** [comment]
Ok - I have updated the save-the-date-description. IMO we could already send it before the date is confirmed, because the description only says "most likely date" - but I'm also ok with waiting till tuesday.

https://www.meetup.com/de-DE/Rust-Cologne-Bonn/events/233139845

**Comment by carlberger** [comment]
Hi,
I spoke to i-dont-know-who at the end of this weeks meeting, suggesting that the rust meetup could be organised in an open-space style.  I think the difference between rust-beginners and highly advanced rust coders at the meeting is a very good thing, and doing such a meeting as an open-space may be worth a shot some times. I just had very good experiences with open-space as a way of doing meetings with lots of people. Interested ? I'd help setting this up, its easy. And I do not have much to say about rust anyway at this stage :) 
cheers,
carl

**Comment by colin-kiegel** [comment]
Hi Carl, you spoke with Florian (@Florob, long brown hair) and me, Colin (reddish short hair). :-)

What a nice coincidence you already found the way to this discussion before we send any Email.

We are generally interested in trying different concepts from time to time to add variety. I think we should look into that idea. But I didnt have time to read about it yet. Its definetely good to know that you would be willing to help. Thank you very much.

Do you think it can be combined with having 1-2 talks first (let's say each about 20-30 minutes) or should we better reserve the whole evening for open spaces?

**Comment by carlberger** [comment]
Hi Colin,
sure thats possible. "open space" means basically that people come up with the things they want to talk about on the spot and then everybody joins the topic they are most interested in. The combination of first some talks, then "open space" is no problem. Actually what happend last meeting was sort of open space in the last hour ... only with the "official open space rules" this is kinda guaranted that everybody comes directly to the topic they want to talk about (or just join in to listen to). 
My input in rust is something close to nothing at this point, but to organize an open space is kind of a no brainer. So if you want to try out the "open space" way, I'd love to help you out there. 
carl

**Comment by colin-kiegel** [comment]
@badboy:
- How much time would you like to have for your talk about _rust to asm.js_?
- Do you see the possibility for a hands-on session after your talk? If so, what are the basic requirements for this?
- disclaimer: I have basically no idea about the maturity level of the whole rust to asm.js thing. Could you summarize it very briefly without spoiling to much? ;-)

EDIT: What other activities could we combine with this talk?

**Comment by Florob** [comment]
I've confirmed that we can indeed have the room on the 5th.

**Comment by badboy** [comment]
@colin-kiegel I'm not sure yet and I will probably not have the full material ready. I can try to have a hands-on ready, but I can't promise anything. I'd be happy to talk about the way to emscripten/asm.js/wasm for about 20-30min though.

**Comment by colin-kiegel** [comment]
Hm, I am just wondering which activities we could combine with this Talk. Any suggestions if the hands-on does not work? Or does someone else have enough knowledge to help prepare a hands-on part?

**Comment by killercup** [comment]
Well, if we can't get the rust compiler to write JS, I'd happy to teach you how to write it by hand! 😄  Btw, @colin-kiegel, have you seen @badboy's http://tmp.fnordig.de/rust-derive-builder/channel.html?

On a more serious note, I'm open to suggestions on what to talk about from that list I gave above. Or, we can just set a general theme and discuss it? I think this worked okay for the macros topic (probably with a bit more of a prepared introduction next time).

Alternatively, we do the talk and then the open space thing.

**Comment by badboy** [comment]
What about a general web theme? We have emscripten for the frontend, Diesel for the database stuff and I'm sure we can kick off a discussion around more webby stuff.

**Comment by killercup** [comment]
Sounds good, @badboy.

I think it was @mre who asked about scraping websites a few meetups ago. Maybe we can pick that up? I've been meaning to look into html5ever (et.al.). Otherwise, I'll bore you with diesel internals (and maybe a little demo? 😉).

**Comment by badboy** [comment]
![why not both?](https://media.giphy.com/media/3o85xIO33l7RlmLR4I/giphy.gif)

**Comment by killercup** [comment]
![](https://media.makeameme.org/created/why-not-zoidberg-e6ta2l.jpg)

**Comment by badboy** [comment]
I'm totally fine with you/us/the meetup [preparing something useful for webscraping, others will benefit](https://github.com/rust-community/rustbridge/issues/12).

**Comment by colin-kiegel** [comment]
Wow, the derive_builder example is cool - a bit heavy though (~1 MB). :-)

Personally I find the (w)asm/js thing very fascinating, because it opens a realm of new possibilities. My expectations for a hands-on part would be:
- _after this session you are able to compile hello world (or something alike) from rust to javascript and run it in your browser - bring your own device to test it out live._

I find the diesel topic also also _super hot_. My hope was that we could dedicate **even more** time to diesel and closely related web-frameworks for rust. I imagine a meetup with a focus roughly like http://www.arewewebyet.org/ -> i.e. sharing experiences with these tools.

So basically my suggestion is this:
- next meetup:
  - asm.js + optional hands-on
  - web-scraping
  - optional: some continuation of the macros topic (where it fits into the overall theme)
  - discussion
- 2nd meetup from now:
  - 'arewewebyet' focus, with diesel center-stage
  - perhaps try out "open spaces" to explore the rust-web-tooling in small groups

EDIT: I just have the feeling that the material would be enough for two meetups

**Comment by badboy** [comment]
@colin-kiegel @killercup & I stripped it down, uglified it and then gzipped and it became much smaller!

**Comment by badboy** [comment]
> after this session you are able to compile hello world (or something alike) from rust to javascript and run it in your browser - bring your own device to test it out live.

Well, yeah, if we make the meetup about 4 hours long, I'm sure everyone can compile it :dancer:
But stay tuned, that might still be possible in some way, I have a plan.

**Comment by killercup** [comment]
I would love to hear @badboy talk about asm.js, and I agree on all of @colin-kiegel's points about the _next meetup_—but can we postpone that web focused meetup to, let's say, November? That may sound pretty random at first, my reasoning is this:
- Diesel will get to 1.0 in the near-future (according the Sean), and
- dealing with async I/O in Rust just got a huge upgrade with stuff like [futures](https://github.com/alexcrichton/futures-rs) and [tokio](https://github.com/tokio-rs)—which basically obsoletes half of what people are currently using the build web stuff, and thus
- it will take some time to get to a point where there are enough libraries that make use of these new tools to justify a meetup where we talk about that stuff.

I'd hate to plan a talk about this stuff and then just discuss stuff like the current iteration of Iron (which hasn't changed in the last year AFAICT) and/or say "this will change with async I/O" in every sentence.

**Comment by badboy** [comment]
So your plan would be:

September: asmjs + web scraping + discussion
October: [fill the gap]
November: arewewebyet/diesel/everything webby|async

Correct?

**Comment by killercup** [comment]
@badboy, yeah, my plan is basically:
1. You talk about asm.js in September and it will be glorious
2. ???
3. PROFIT!

I guess step three could also be a web-focused meetup… 😄  Seriously though, that November date was just a guess, as I expect stuff to happen around the time of the conferences; we should have a good idea of/have seen all the shiny new webby goodness that was presented at all of the three conferences in November.

**Comment by badboy** [comment]
http://slides.com/hannelitavante-hannelita/rust-html-parsing#/
https://github.com/galois1/demo/blob/improv1/src/main.rs (more in branches)

**Comment by badboy** [comment]
To mention: 
- new errors (https://github.com/rust-lang/rust/issues/35233, https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)

**Comment by arnecls** [comment]
Hey guys, I already spoke to Colin but he pointed me to here.
I would love to hold a talk about memory management at your usergroup. I once had a lightning talk in the PHP usergroup in Düsseldorf on "Why I hate garbage collectors" and I would like to deepen this a little bit more. Going from what I think are the drawbacks of GC as well as manual memory management to why RAII or the borrow / lifetime concept of Rust is a very valid alternative for realtime programming (disclaimer: I have not yet written a single line of Rust, but I follow it's development). I draw from ~10 years of experience on that topic, so there is a lot to tell.

If you are interested, please notify me, otherwise I will probably just join as a listener ;)

**Comment by badboy** [comment]
Hey @arnecls. That sounds interesting indeed. How long would your talk be?

@Rustaceans/organizers-cologne: What do the others think? Should we do the two talks (arne's and mine) in September and have the more interactive meetup around web stuff in October or November?

**Comment by colin-kiegel** [comment]
@badboy If you need time for the 'hands-on compiling to asm.js' part, I would definitely prioritize that for the next meetup - if that is ok for you. If you think we will have spare time, we could either fill it with the 'web scraping' suggestion or 'drawbacks of GC' (but probably not both). That would be my line of reasoning right now.

Anyone from our meetup group reading this may also state his or her _personal preference_ about these choices. If we get substantial feedback (i.e. more than one person) within the next days, we could take this into consideration. :-)

**Comment by badboy** [comment]
Yeah, not web scraping and GC then.
My plan would be:
- 30min GC
- ~20min asmjs talk + hands-on

That would fit time-wisely.
Moving GC to a later date would be fine as well, then we would have asmjs + webscraping for september.

**Comment by arnecls** [comment]
The original talk was 10min. I guess 30min is totally ok for an extended version.
My October schedule is already pretty packed, so if you want to schedule it for a later meetup November will be my next free slot.

**Comment by badboy** [comment]
Ok. Even 10 or 20minutes should be fine.

**Comment by colin-kiegel** [comment]
what about 10 minutes + discussion? This way @arnecls does not need to change his talk much and it might be more interactive. Please only extend a talk if you have the feeling this is beneficial - not for the purpose to fill a time slot. If we happen to have _too much_ time, we can still make some breaks here and there and engage in spontaneous and informal talking.. :-)

**Comment by arnecls** [comment]
The original talk is quite old and was designed to be a bit controversial.
So I would want to extend or change it anyway to give a little bit more insight on the whys.
20 minutes would be most comfortable I guess.

**Comment by colin-kiegel** [comment]
@Rustaceans/organizers-cologne we could already prepare the final invitation, _if_ we agree on this agenda.

I prepared a draft in an empty new branch of this repository:
- https://github.com/Rustaceans/rust-cologne/blob/drafts/2016-09-05_meetup-invitation.md 

Feel free to edit the draft as you like. :-)
I think this is better than posting links to world-public etherpads here...

**Comment by colin-kiegel** [comment]
PS: @arnecls you might also want to have a look if the working title for your talk is ok. ;-)

**Comment by killercup** [comment]
@arnecls, I have to admit, I'm pretty skeptic towards talks whose titles start with "Why I hate…" – but I'm looking forward to the discussion that will surely follow it! 😄

> I prepared a draft in an empty new branch of this repository

The invite text sounds good (skipping the part you want @badboy to add).

Is the hands-on part doable? I.e., is there a rustup toolchain that actually works with asm.js right now that mabye @badboy has on a server somewhere? (Waiting for rustbuild is not something I'd consider a group activity…)

> Feel free to edit the draft as you like. :-)
> I think this is better than posting links to world-public etherpads here...

@colin-kiegel I'm not really concerned with posting those links in public, but sure, this works as well. We could also use Github's wiki, which is basically just another branch/sub-repo full of markdown anyway.

**Comment by colin-kiegel** [comment]
@killercup I am not concerned about people messing with our text - I am just cautious about the next generation of bots using jan-eriks pads for spam or to control a botnet or whatever. :-)

I share your concerns about a strongly negative title - it could shift the discussion climate into a direction we don't want. @arnecls could you please suggest another title, which is either _plain positive_ or _just objective_?

**Comment by killercup** [comment]
I wouldn't change the title, that would be dishonest IHO. I enjoy a good
rant, and am known to rant about lots of stuff –just be sure the content is
a based on facts and that you don't insult people ('insulting'
technologies/implementations by disproving their usefulness for their use
cases is okay in my book).

Re: discussion climate: We should (and always should've) mention the CoC
and be ready to moderate the discussion.

Colin Kiegel notifications@github.com schrieb am Sa. 13. Aug. 2016 um
16:28:

**Comment by arnecls** [comment]
The talk is far less controversial than the title is, so I don't think the discussion will go into a negative direction. I currently don't have another title that does not make people fall asleep as soon as they read it or sounds just plain confusing. 
- "Drawbacks of garbage collection over (semi) automatic resource management" ... meh
- "GC vs RAII" ... wtf is RAII?
- "Garbage Collection vs Resource Acquisition is Initialization" ... meh
- "Evolution of memory management" ... nah, not enough historic context

... I'll try to come up with an alternative title.

**Comment by colin-kiegel** [comment]
@arnecls it's alright - we can keep it. With both your explanation and killercups last comment all trouble is gone now. ^^

**Comment by badboy** [comment]
It's definitely not possible to build the toolchain on-site. I can take a look if I can pre-compile it, it will be _only_ for a Linux machine then of course.
But what I actually want to have is a playground offering the emscripten-option, need to work on that.

**Comment by killercup** [comment]
Or you can go all 2016 on this issue and make a Docker image ;)

Jan-Erik Rediger notifications@github.com schrieb am So. 14. Aug. 2016 um
16:55:

**Comment by badboy** [comment]
Good point. Never really used Docker, but that shouldn't be too hard

**Comment by colin-kiegel** [comment]
@badboy sounds good. I guess we still wait with the invitation until the requirements are clear and we know whether the hands-on is going to happen, right?

**Comment by badboy** [comment]
We got a Docker image! Though it's 3G big, but that should still be possible to try for those familiar with Docker.
I also have a somewhat working Playground, though it is limited in what it can do.

All in all this means I can provide a very basic hands-on, but it will not fill the evening (only for those want to hack around).

**Comment by colin-kiegel** [comment]
Ok, cool. I am very excited about this. :-)

Do you know how you want to distribute the image? USB-Sticks and/or via local fileserver?

If we have some time left, I could imagine brainstorming as a group about activities and topics of future meetups. The last time we did this has been a while ago. This way we could generate some ideas especially for the next meetup. We can keep this optional and decide spontaneously whether we want to do this.

@badboy are you ok with us sending the invitation? The draft is here https://github.com/Rustaceans/rust-cologne/blob/drafts/2016-09-05_meetup-invitation.md (you can edit it via browser - it still has a gap at the exact requirements for the hands-on).

**Comment by badboy** [comment]
@colin-kiegel local webserver & usb stick & docker hub should be a good way to hand it out.
Will edit my part of the invitation in the next hour

**Comment by colin-kiegel** [comment]
@Rustaceans/organizers-cologne Great - I just send the main invitation via meetup. Anyone who likes could do some of our other channels and check it in the expanded checklist in this post: https://github.com/Rustaceans/rust-cologne/issues/6#issue-169238248 :-)

**Comment by colin-kiegel** [comment]
Just did the calendar and forums - reddit/twitter/website are still left.

(mh.. not much happening at rustplatz anymore.. my post was the first since 6days!)
- https://calendar.google.com/calendar/render?eid=NGk2ZG80cTNldDlpbWRzcGtrM3VxdTc5bzggYXBkOXZtYmMyMmVnZW5tdHU1bDZjNWpiZmNAZw&ctz=America/Los_Angeles&sf=true&output=xml#eventpage_6
- https://forum.rustplatz.de/t/koeln-10-compiling-rust-to-your-browser-montag-2016-09-05/161
- https://users.rust-lang.org/t/cgn-10-compiling-rust-to-your-browser-monday-2016-09-05/6916

**Comment by killercup** [comment]
- Voilà: http://rustaceans.cologne/2016/09/05/compile-to-js.html
- ~~Will drop a tweet in a few minutes.~~ [Tweet](https://twitter.com/RustCologne/status/765826101058863104)
- [Reddit](https://www.reddit.com/r/rust/comments/4y4iq6/meetup_in_cologne_on_20160905_about_compiling/)

(Remind me to give the twitter credentials to the other @Rustaceans/organizers-cologne as well!)

**Comment by Florob** [comment]
Do we want to record talks this time around? In particular @arnecls and @badboy do you have any interest in being recorded? I think it's nice for the german community and not too much overhead to do, but I'll defer to other opinions.

**Comment by badboy** [comment]
If we have the equipment and you're okay with it, we can do it. I won't have the time to make this my best talk ever, but I'll sure prepare it and so I'm fine with being recorded (if I really fail at it we can still decide not to publish :D)

**Comment by badboy** [comment]
As we wanted to make this, here's a list of things I'd like to mention at the beginning:

These are all related to current community efforts:
- RustBridge - project to make the Rust Language more accessible to people who are underrepresented in technology. Initial work on workshop material has happened. All help appreciated.
- bors/homu - people want to bring it back being open to the community. They struggle with the setup and maintaining. Help appreciated

**Comment by killercup** [comment]
@badboy, should we collect these things somewhere semi-permanently? A general "Info" page on the website (i.e., a markdown file in this repo) might be nice, with links to the CoC, thanks the location sponsor(s), and includes generally useful links to community efforts like these.

Also, I don't often stalk Alex on GitHub, but when I do, I find promising repos like https://github.com/alexcrichton/bors2 😄

**Comment by killercup** [comment]
The only thing left open in the list above is "Agenda planning." Is the plan to have two talks and a hands-on enough of an agenda? Then I think we are good to go here! :)

We might want to add some more "promotion times", though, to remind people of the event. I'll try to write a few more tweets. I was going for mid-day today, some time on the weekend (Sat/Sun evening to get weekend hackers), and mid-day on Monday. I'm open for suggestions for other times, though!

**Comment by badboy** [comment]
Agenda: ✓

**Comment by colin-kiegel** [comment]
I am looking forward to it :-)

**Comment by killercup** [comment]
What's that I [heard](https://twitter.com/badboy_/status/772400032935010304)/read (https://github.com/Rustaceans/rust-cologne/issues/6#issuecomment-240431331) about recording @badboy's talk? Will this actually happen? I assume we want that part of/the meetup to be in English, then?

**Comment by killercup** [comment]
Some notes.

## Stuff to mention at the event

### Meta
- Where/how to get drinks, donate to C4
- CoC
- Idea: Open space? (@carlberger)
- Rustfest Rustfest Rustfest
- Kathrine from Honeypot asked via meetup.com if we could link to their developer survey

## News
- [Rust 1.11](https://blog.rust-lang.org/2016/08/18/Rust-1.11.html)
- Truckloads of tokio/futures news (tl;dr wait for more blog posts/things to settle down)
- [docs.rs](https://docs.rs) now automagically renders docs for everything on crates.io!
- new wonderful errors (https://github.com/rust-lang/rust/issues/35233, https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)
- news regarding the 2nd edition of The Rust Programming Language: http://words.steveklabnik.com/whats-new-with-the-rust-programming-language
- https://github.com/colin-kiegel/rust-derive-builder
- Cargo: Add support local mirrors of registries https://github.com/rust-lang/cargo/pull/2857 (probably still nightly only)

**Comment by colin-kiegel** [comment]
Thx everyone for preparing this great meetup :-)

We will upload the slides and recordings to our website the next days:
http://rustaceans.cologne/2016/09/05/compile-to-js.html

We already collect ideas for our next meetup in #7
